examples/call_go_from_prolog_2/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/ichiban/prolog"
+	"github.com/ichiban/prolog/engine"
+)
+
+func main() {
+	p := prolog.New(nil, nil)
+
+	// Define a custom predicate of arity 1: get_time(-Timestamp)
+	p.Register1(engine.NewAtom("get_time"), func(_ *engine.VM, TimeStamp engine.Term, k engine.Cont, env *engine.Env) *engine.Promise {
+		// Check if the input arguments are of the types you expected.
+		_, ok := env.Resolve(TimeStamp).(engine.Variable)
+		if !ok {
+			return engine.Error(engine.TypeError(engine.NewAtom("variable"), TimeStamp, env))
```

Comment:
They removed `type_error(variable, Culprit)` from the standard in ISO/IEC 13211-1:1995/Cor.2:2012 and replaced it with `uninstantiation_error(Culprit)`.

see https://www.complang.tuwien.ac.at/ulrich/iso-prolog/error_k#error_classes

Another predicate design I often see is to think `Timestamp` can be an integer but you can leave it as a variable. Let's say we call `get_time(1735657200)` at the right time. It should succeed, right? If you missed the timing, it fails. A variable should work fine too but for anything else, `type_error(integer, Culprit)`.

You don't have to follow these in your `get_time/1` but I prefer the ISO Prolog way in the examples.

Reply:
@ichiban, afais, the` uninstantiation_error(Culprit)` error term isn’t available in the repo. Do you think it would be a useful addition?

Reply:
@ccamel ~It's not implemented bc I didn't need it while implementing ISO predicates.~ Oh, wait. I should have thrown `uninstantiation_error(Stream)` in open/[3, 4]. https://github.com/ichiban/prolog/issues/339

Yes, it would be a great addition!